File: problems_051to075/problem_073/solution.py
import os
import sys
from math import ceil, floor, gcd

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 0

        for denominator in range(2, 12000+1):
            max_numerator = floor(denominator * 1 / 2)

            if max_numerator == denominator * 1 / 2:
                max_numerator -= 1

            min_numerator = ceil(denominator * 1 / 3)

            if min_numerator == denominator * 1 / 3:
                min_numerator += 1

            for numerator in range(min_numerator, max_numerator+1):
                if gcd(numerator, denominator) == 1:
                    answer += 1

        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_051to075/problem_073/solution.ts
import { SolutionFunction } from '@/lib/types';
import { gcd } from '@/lib/toolbox';

const solution: SolutionFunction = () => {
  let answer: ReturnType<SolutionFunction> = 0;

  for (let denominator = 2; denominator <= 12000; denominator += 1) {
    let maxNumerator = Math.floor(denominator / 2);

    if (maxNumerator === denominator / 2) {
      maxNumerator -= 1;
    }

    let minNumerator = Math.ceil(denominator / 3);

    if (minNumerator === denominator / 3) {
      minNumerator += 1;
    }

    for (let numerator = minNumerator; numerator <= maxNumerator; numerator += 1) {
      if (gcd(numerator, denominator) === 1) {
        answer += 1;
      }
    }
  }

  return answer;
};

export default solution;
